### PR TITLE
Fix false positive in bugprone-throw-keyword-missing

### LINKED ISSUE #115055: False positive bugprone-throw-keyword-missing with NSDMI
Example:

```cpp
struct RegularException
{
    RegularException(int);
};

class Foo
{
 public:
    Foo(){} 

 private:
    RegularException x_{123};
};

```

Gives:
```console
[source>:12:24: warning: suspicious exception object created but not thrown; did you mean 'throw RegularException'? [bugprone-throw-keyword-missing]](javascript:;)
   12 |     RegularException x_{123};
      |                        ^
```

[Godbolt](https://godbolt.org/z/47rsz97hf)

### PR DESCRIPTION
**Pull request by carlosgalvezp**
Fixes #115055

### CODE REVIEW

**Comment by carlosgalvezp** [comment]
Ok it seems the Windows build is broken in general, please disregard that!

**Review by 5chmidti — APPROVED** [review]
LGTM

---

I filed #115540 because I noticed how the `hasAncestor` matcher in the `unless` part means that we are excluding quite some cases. (unrelated to this PRI)

File: clang-tools-extra/test/clang-tidy/checkers/bugprone/throw-keyword-missing.cpp [hunk 1]
```diff
@@ -139,6 +139,14 @@ CtorInitializerListTest::CtorInitializerListTest(float) try : exc(RegularExcepti
   RegularException();
 }
 
+// https://github.com/llvm/llvm-project/issues/115055
```
Comment (HerrCai0907) [inline clang-tools-extra/test/clang-tidy/checkers/bugprone/throw-keyword-missing.cpp:142]: use a namspace GHxxxx (issue number)

File: clang-tools-extra/test/clang-tidy/checkers/bugprone/throw-keyword-missing.cpp [hunk 2]
```diff
@@ -139,6 +139,14 @@ CtorInitializerListTest::CtorInitializerListTest(float) try : exc(RegularExcepti
...
+class CtorInitializerListTest2 {
+ public:
+  CtorInitializerListTest2(){}
```
Comment (HerrCai0907) [inline clang-tools-extra/test/clang-tidy/checkers/bugprone/throw-keyword-missing.cpp:145]:
```suggestion
  CtorInitializerListTest2() {}
```

**Review by HerrCai0907 — APPROVED** [review]
lgtm

**Review by PiotrZSL — APPROVED** [review]

**Pull request merged**